hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java

```diff
@@ -1810,4 +1810,10 @@
 	 *
 	 */
 	String TABLE_GENERATOR_STORE_LAST_USED = "hibernate.id.generator.stored_last_used";
+
+	/**
+	 * Raises an exception when an in-memory intensive operation is about to be performed.
+	 * Disabled by default. Set to true to enable.
+	 */
+	String FAIL_ON_INTENSIVE_IN_MEMORY_OPERATION = "hibernate.fail_on_intensive_in_memory_operation";
```

Comment:
IMO this name is way too vague.  I'd prefer to name this something more concrete, like `hibernate.query.fail_on_pagination_over_collection_fetch`

Reply:
Done